src/controller/CHIPCommissionableNodeController.cpp

```diff
@@ -41,5 +44,20 @@ const Mdns::DiscoveredNodeData * CommissionableNodeController::GetDiscoveredComm
     return GetDiscoveredNode(idx);
 }
 
+CHIP_ERROR CommissionableNodeController::AdvertiseCommissionableNode()
+{
+#if CONFIG_DEVICE_LAYER
+    ReturnErrorOnFailure(chip::Mdns::ServiceAdvertiser::Instance().Start(&chip::DeviceLayer::InetLayer, chip::Mdns::kMdnsPort));
+#endif
+    return app::Mdns::AdvertiseCommissionableNode();
```

Comment:
does it make sense to call this if the above CONFIG_DEVICE_LAYER is 0?

Reply:
No. Added an #else block.